app/lib/task/scheduler.dart

```diff
@@ -307,14 +307,50 @@ Future<Payload?> updatePackageStateWithPendingVersions(
     s.derivePendingAt();
     tx.insert(s);
 
+    final payloadVersions = pendingVersions
```

Comment:
Might be a tiny bit easier if we sort `pendingVersions` before doing the `.map`.

Also then we won't need: `VersionTokenPair.semanticVersion`.